src/Runtime/OMTensor.inc

```diff
@@ -113,20 +121,17 @@ OMTensor *omTensorCreate(
     return NULL;
   if ((tensor->_shape = (int64_t *)malloc(rank * sizeof(int64_t))) &&
       (tensor->_stride = (int64_t *)malloc(rank * sizeof(int64_t)))) {
-    // If malloc for _shape or _stride fails, free them and return NULL.
-    if (!tensor->_shape || !tensor->_stride) {
-      if (tensor->_shape)
-        free(tensor->_shape);
-      if (tensor->_stride)
-        free(tensor->_stride);
-      return NULL;
-    }
     tensor->_allocatedPtr = data_ptr;
     tensor->_alignedPtr = data_ptr;
     tensor->_rank = rank;
     tensor->_dataType = dtype;
     tensor->_owning = false;
+  } else {
+    if (tensor->_shape)
```

Comment:
Smart to test only the first one... as if the first one fails, then the second pointer is garbage, and if the second one fails, then we have to free only the first one. 

Why are some failure to allocate memory resulting here in returning NULL, whereas in `OMTensor(int rank)` above we throw an assert?

Also, do you check if pointers are not null before usage? If so, you may want to set the pointers (such as `tensor->_shape` back to null in such cases.

Reply:
`OMTensor(int rank)` is a C++ constructor which does not return value so it throws a runtime_error.

`tensor->_shape` and `tensor->_stride` are guaranteed to be not NULL if a tensor is created successfully, i.e., they are always initialized upon successful tensor creation so there is no need to set them to NULL upon failure.

Actually I think I just saw a bug in `omTensorDestroy` that `tensor->_shape` and `tensor->_stride` were not freed. I will fix that. Thanks.